ramips: Reorganize device and DTS names (like ath79)

adschm opened the pull request:
This aims at reorganizing ramips to introduce a device naming scheme into ramips that is similar to ath79:
- Name devices in Makefile as in the DTS compatible
- Name DTS files like soc_vendor_device.dts
- Calculate DTS names from device names
- Fix some inconsistencies on the way

**This is ready for review and has been run-tested on several subtargets.**

[Tested on rt305x/mt7620/mt7621/mt76x8]
Tested-by: INAGAKI Hiroshi &lt;musashino.open@gmail.com&gt;

_This PR now holds the final rebased version. For the old sequence of patches with fixes at the end, please use:_
https://github.com/adrianschmutzler/openwrt/commits/ramipsfix

---

Stuff below is outdated and only for reference:

ToDo:
- ~~Introduce device renames into all those board.d etc. files (nothing done for current blue print)~~ This is included for the converted targets now (done by scripts and then checked manually)
- ~~Do subtargets besides mt7620/mt7621~~
- Stuff I overlooked
- ~~Sort base-files and Makefiles again~~
- ~~Sort Makefiles~~

Issues:
- ~~dlink_dwr-921-c1 and dlink_dwr-921-c3 share the DTS of dlink_dwr-921-c1.~~

Questions:
- ~~Use mt7620 or mt7620a/mt7620n for SOC?~~ I will use mt7620a/n, mt7628an, etc. now

981213 commented on target/linux/ramips/image/Makefile:
```diff
@@ -21,10 +21,12 @@ KERNEL_LOADADDR := $(loadaddr-y)
 
 KERNEL_DTB = kernel-bin | patch-dtb | lzma
 define Device/Default
+  ATH_SOC :=
```
ATH actually means Atheros. You might want RT_SOC (RT for Ralink) or MT_SOC (MT for Mediatek, who acquired Ralink) here.

981213 replied:
And a new variable in device template should be added to DEVICE_VARS.

adrianschmutzler replied:
Oh..... Yes.

musashino205 replied:
I suggest ```MTK_SOC```.

981213 replied:
Oops...
Variables in DEVICE_VARS will be cleared for every single device profile. If it's defined at the beginning of {subtarget}.mk, it might not be a good idea to put ATH/RT/MT_SOC in device profile.
This may need to be defined for every devices since we have mt7620a/mt7620n in mt7620, mt7628/mt7688 in mt76x8 and so on.

adrianschmutzler replied:
Per-device would then be the same as for ath79.
This looks a little undesirable to me (as subtargets are already "sorted"), but I do not see a way around yet.

adrianschmutzler replied:
I've now put MTK_SOC in device profiles for all subtargets. This is easy to achieve programatically, and if someone finds a way to specify a common subtarget definition, it would also be easy to remove again.

981213 reviewed (commented):
Ah... It seems that this one conflicts with https://github.com/openwrt/openwrt/pull/2170 :(

981213 commented:
I'll rebase #2170 onto this PR after you finished it :) It's easier than rebasing this PR onto #2170 .

981213 commented on target/linux/ramips/image/mt7620.mk:
```diff
@@ -131,6 +131,7 @@ define Device/bocco
   DTS := BOCCO
   DEVICE_TITLE := YUKAI Engineering BOCCO
   DEVICE_PACKAGES := kmod-sound-core kmod-sound-mt7620 kmod-i2c-ralink
+  SUPPORTED_DEVICES += planex,cs-qr10
```
You don't actually need this. The first compatible string is used as board name only if it's not specified in target/linux/ramips/base-files/lib/ramips.sh

981213 replied:
Ah... This problem exists in all your current changes....

981213 commented on target/linux/ramips/image/mt7620.mk:
```diff
@@ -127,52 +127,52 @@ define Device/bdcom_wap2100-sk
+TARGET_DEVICES += yukai_bocco
 
-define Device/c108
+define Device/hnet_c108
```
However when doing these renamings you'll need to add SUPPORTED_DEVICES as the default one in target/linux/ramips/image/Makefile may not match board name in base-files/lib/ramips.sh.

adrianschmutzler replied:
Ah, I now realize that I had a wrong understanding of how the board name for upgrade was derived. I thought it was based on the compatible, as in ath79 the compatible and device name match each other (except , vs _).
I will have to think about the best way to address this.

981213 commented:
> dlink_dwr-921-c1 and dlink_dwr-921-c3 share the DTS of dlink_dwr-921-c1.

DTS can still be overrided in device profile even if it's specified in profile template. You could just override DTS variable for dlink_dwr-921-c3.

981213 commented on target/linux/ramips/image/mt7620.mk:
```diff
@@ -5,6 +5,8 @@
 DEVICE_VARS += TPLINK_FLASHLAYOUT TPLINK_HWID TPLINK_HWREV TPLINK_HWREVADD TPLINK_HVERSION \
 	DLINK_ROM_ID DLINK_FAMILY_MEMBER DLINK_FIRMWARE_SIZE DLINK_IMAGE_OFFSET
 
+ATH_SOC := mt7620
```
You can add a Device/Default in this file and override this one in default device profile template instead.

981213 replied:
Ah... I'm not sure if what I've said above actually works.

adrianschmutzler replied:
I wonder whether mt7620 is enough here or whether one would distinguish between those mt7620a and mt7620n (which would require at least some device-specific SOC variables as in ath79 again)

adrianschmutzler commented:
> Ah... It seems that this one conflicts with #2170 :(

Rebasing will be doable. Since this one will take its time, you might have to wait some time. If you find some testers, I wouldn't complain if your's is first.

adrianschmutzler commented:
Okay, so let me know whether I got it right (board name during upgrade check):

On the device I want to update, the current board is retrieved from /tmp/sysinfo/board_name.
It gets there
1. from all entries in base-files/lib/ramips.sh
2. from the device compatible if no entry in ramips.sh exists

For upgrade, this is compared with the image to be flashed, where the SUPPORTED_DEVICES variables is used.
This one is initialized to $(subst _,$(comma),$(1)), where $(1) is the Device node name (e.g. ArcherC50v1 from Device/ArcherC50v1) in the Makefile. In the subtarget Makefiles this value can be overwritten (:=) or other values can be added (+=) per device.

So, if I change the node name in the Makefiles (ArcherC50v1 to tplink_archer-c50-v1), I will have to add the old node name in SUPPORTED_DEVICES (with +=). However, if I have to add SUPPORTED_DEVICES anyways, I could then also remove the entry in ramips.sh, so the DTS compatible is used (which will most probably empty ramips.sh in the end).
This will have no sideeffects except that downgrades will need -F then.

Did I get it right?

adrianschmutzler commented on target/linux/ramips/image/mt7620.mk:
```diff
@@ -658,6 +658,7 @@ TARGET_DEVICES += edimax_ew-7478apc
 define Device/tplink_c2-v1
   $(Device/Archer)
   DTS := ArcherC2-v1
+  SUPPORTED_DEVICES += tplink,c2-v1
```
required, also needs rename of device

adrianschmutzler commented on target/linux/ramips/image/mt7620.mk:
```diff
@@ -81,7 +81,7 @@ endef
   $(Device/Archer)
   DTS := ArcherC20i
-  SUPPORTED_DEVICES := c20i
+  SUPPORTED_DEVICES := c20i tplink,c20i
```
not required

adrianschmutzler commented on target/linux/ramips/image/mt7620.mk:
```diff
@@ -670,6 +671,7 @@ TARGET_DEVICES += tplink_c2-v1
 define Device/tplink_c20-v1
   $(Device/Archer)
   DTS := ArcherC20v1
+  SUPPORTED_DEVICES += tplink,c20-v1
```
required, also needs rename of device

adrianschmutzler commented on target/linux/ramips/image/mt76x8.mk:
```diff
@@ -270,6 +270,7 @@ define Device/tplink_c20-v4
   TPLINK_HWREVADD := 0x4
   TPLINK_HVERSION := 3
   DEVICE_PACKAGES := kmod-mt76x0e
+  SUPPORTED_DEVICES += tplink,c20-v4
```
required, also needs rename of device

adrianschmutzler commented on target/linux/ramips/image/mt7620.mk:
```diff
@@ -95,7 +95,7 @@ TARGET_DEVICES += ArcherC20i
   $(Device/Archer)
   DTS := ArcherC50
-  SUPPORTED_DEVICES := c50
+  SUPPORTED_DEVICES := c50 tplink,c50
```
not required

adrianschmutzler commented on target/linux/ramips/image/mt76x8.mk:
```diff
@@ -284,6 +285,7 @@ define Device/tplink_c50-v3
   TPLINK_HWREVADD := 0x1
   TPLINK_HVERSION := 3
   DEVICE_PACKAGES := kmod-mt76x2
+  SUPPORTED_DEVICES += tplink,c50-v3
```
required, also needs rename of device

adrianschmutzler commented on target/linux/ramips/image/mt76x8.mk:
```diff
@@ -299,6 +301,7 @@ define Device/tplink_c50-v4
   TPLINK_HVERSION := 3
   DEVICE_PACKAGES := kmod-mt76x2
   IMAGES := sysupgrade.bin
+  SUPPORTED_DEVICES += tplink,c50-v4
```
required, also needs rename of device

adrianschmutzler commented on target/linux/ramips/image/mt7620.mk:
```diff
@@ -110,7 +110,7 @@ TARGET_DEVICES += ArcherC50v1
   $(Device/Archer)
   DTS := ArcherMR200
-  SUPPORTED_DEVICES := mr200
+  SUPPORTED_DEVICES := mr200 tplink,mr200
```
not required

adschm reviewed (commented):
Reminder for next update.

adrianschmutzler commented on target/linux/ramips/image/mt7620.mk:
```diff
@@ -257,6 +239,7 @@ TARGET_DEVICES += dlink_dwr-921-c1
 
 define Device/dlink_dwr-921-c3
   $(Device/dlink_dwr-921-c1)
+  DTS := mt7620_dlink_dwr-921-c1
```
In parent Makefile DTS is set up with `DTS = ...`
Will I need `=` or `:=` here to overwrite that?

981213 replied:
Both of them should work. The difference between = and := is that whether the expression after it is expanded before or after the variable is used. This is just a plain string and there is nothing to expand anyway.

adrianschmutzler commented:
Adressed the first bunch of comments except the MTK-SOC/ATH-SOC in mt7620.mk.

Still draft, so only mt7620 and no board.d adjustments for device renaming.

981213 commented:
> So, if I change the node name in the Makefiles (ArcherC50v1 to tplink_archer-c50-v1), I will have to add the old node name in SUPPORTED_DEVICES (with +=). However, if I have to add SUPPORTED_DEVICES anyways, I could then also remove the entry in ramips.sh, so the DTS compatible is used (which will most probably empty ramips.sh in the end).

And you'll then need to change board names in base-files/etc/{uci-defaults,board.d}/ and do a ton of reordering like what @pepe2k did a year ago :)

> This will have no sideeffects except that downgrades will need -F then.
> 
> Did I get it right?

Correct.

adrianschmutzler commented:
> And you'll then need to change board names in base-files/etc/{uci-defaults,board.d}/ and do a ton of reordering like what @pepe2k did a year ago :)

Indeed, however this can be done mostly with scripts.

With the current state of mixed naming conventions, ordering is pretty useless anyway.
I plan to do sorting at the very end, also supported by scripts.

Still, the more I deal with this, the more I desire to clean it up eventually.

adrianschmutzler commented:
Added mt7621 and full change of all base-files for mt7620 and mt7621.

adrianschmutzler commented:
Okay, everything should be complete now except for sorting the Makefiles (I will only do that after everything is ACKed).
I will do some build-testing during the next days ...

adrianschmutzler commented on target/linux/ramips/dts/mt7620a_hiwifi_hc5661.dts:
```diff
@@ -1,6 +1,6 @@
 /dts-v1/;
 
-#include "HC5X61.dtsi"
+#include "mt7620a_hiwifi_hc5x61"
```
.dtsi missing
(Spotted during build)

adrianschmutzler commented on target/linux/ramips/dts/mt7620a_hiwifi_hc5761.dts:
```diff
@@ -1,6 +1,6 @@
 /dts-v1/;
 
-#include "HC5X61.dtsi"
+#include "mt7620a_hiwifi_hc5x61"
```
.dtsi missing
(Spotted during build)

adrianschmutzler commented on target/linux/ramips/dts/mt7620a_hiwifi_hc5861.dts:
```diff
@@ -1,6 +1,6 @@
 /dts-v1/;
 
-#include "HC5X61.dtsi"
+#include "mt7620a_hiwifi_hc5x61"
```
.dtsi missing
(Spotted during build)

adrianschmutzler commented:
Just finished build testing on all six subtargets. The only small inconsistency has been fixed in the latest commit. I also did a quick shellcheck on the changed base-files files.

I would be happy if someone run-tested this, although I do not think this has to be done extensively or on many devices.
Despite, I'd say this is ready for review.

musashino205 commented:
@adrianschmutzler I tested this PR on Buffalo WSR-1166DHP.
mt7621_buffalo_wsr-1166dhp.dts (old: WSR-1166.dts) has a compatible ```buffalo,wsr-1166dhp```, but ```wsr-1166``` is used for device identification in *02_network* before and it's ```buffalo,wsr-1166``` now. So it does not seem to match.

#### mt7621_buffalo_wsr-1166.dts
```
/ {
	compatible = "buffalo,wsr-1166dhp", "mediatek,mt7621-soc";
	model = "Buffalo WSR-1166DHP";
```

#### 02_network
```
	buffalo,whr-300hp2|\
	buffalo,whr-600d|\
	buffalo,wsr-1166|\     <---
	buffalo,wsr-600)
		ucidef_add_switch "switch0" \
			"0:lan:1" "1:lan:2" "2:lan:3" "3:lan:4" "4:wan:5" "6@eth0"
		;;
```

This problem may also occur in WSR-600DHP.

adrianschmutzler commented:
@musashino205 Thanks for the report.
Indeed, I seem to have overlooked the "dhp" in the compatible. All other changes seem to be consistent then (everything is renamed to the same wrong new device name, just not matching the compatible).

I will write a small script to check that for all DTSes, will report later this day and maybe fix others if they occur.

adrianschmutzler commented:
From 316 devices, 4 were wrong (1.3 %) and you spotted 2 (50 %) with one test.
Maybe you should work at QA somewhere ... ;-)

I've fixed all of them and pushed just now (commits "Errata 2", so no rebasing/squashing yet).

I assumed that the DHP suffix is supposed to be added, as one might also have removed it from the compatible without changing the rest.

adrianschmutzler commented:
Argh... Just found that the buffalo devices also need uboot-envtools adjustments. Wait for another fix ...

adrianschmutzler commented:
Added latest fix.
A rebased variant (with all Errata squashed were they belong) is available at https://github.com/adrianschmutzler/openwrt/commits/ramipsfinal

adrianschmutzler commented:
@lynxis Maybe have a look at this if you plan to merge other ramips devices.

adrianschmutzler commented:
I've updated/rebased my tree at https://github.com/adrianschmutzler/openwrt/commits/ramipsfinal to include the recently added TL-WR841N v14 (only DTS rename takes place there).

I will keep the tree in this PR as it is for now, so testers/reviewers can still use it.

adschm committed:
ramips: Improve compatible for TP-Link Archer devices

Include "Archer" in compatible as it is part of the device name.

Update Makefile device names where necessary to match compatible.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adschm committed:
ramips/mt7620: Synchronize Makefiles with DTS compatible

This will "rename" devices in Makefiles to the pattern used in
DTS compatible. This will systematize naming of devices
enormously.

As device names are used to for default SUPPORTED_DEVICES entries,
we need to adjust the source for /tmp/sysinfo/board_name, too.
So remove relevant entries from base-files/lib/ramips.sh and
use device compatible for that.

Despite that, base-files are updated, too.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adschm committed:
ramips/mt7621: Synchronize Makefiles with DTS compatible

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adschm committed:
ramips/mt76x8: Synchronize Makefiles with DTS compatible

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adschm committed:
ramips/rt288x,rt3883: Synchronize Makefiles with DTS compatible

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adschm committed:
ramips/rt305x: Synchronize Makefiles with DTS compatible

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adschm committed:
ramips: Derive DTS name from device name in Makefile

This will automatically derive the DTS name as in ath79 and thus
makes specifying DTS for every device obsolete.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

---

This patch only introduces the mechanism and is then followed by
commits with renames and Makefile adjustments per subtarget.

Eventually, those can be all squashed into a single commit or left
as they are to enhance overview.

adschm committed:
ramips/mt7620: Name DTS files based on scheme

As introduced with ath79, DTS files for ramips will now be labelled
soc_vendor_device.dts(i). With this change, DTS files can be
selected automatically without further manual links.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adschm committed:
ramips/mt7621: Name DTS files based on scheme

As introduced with ath79, DTS files for ramips will now be labelled
soc_vendor_device.dts(i). With this change, DTS files can be
selected automatically without further manual links.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adschm committed:
ramips/mt76x8: Name DTS files based on scheme

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adschm committed:
ramips/rt288x,rt3883: Name DTS files based on scheme

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adschm committed:
ramips/rt305x: Name DTS files based on scheme

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adschm committed:
ramips: Remove base-files/lib/ramips.sh

Having converted the target to use device compatible, ramips.sh
is obsolete now.

The only remaining entry for the mt7688 evaluation board seems to
be orphaned.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adschm committed:
uboot-envtools: Apply ramips device renames

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adschm committed:
kernel/om-watchdog: Apply device renames from ramips

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adschm committed:
ramips: Reapply sorting to board.d files

This patch is cosmetical:
It only applies sorting based on device names, whereas renames
already took place in earlier patches.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

musashino205 commented:
Run tested on:
- rt305x
Buffalo WHR-G300N (rt305x)

- mt7620
Buffalo WHR-1166D (mt7620)
Buffalo WHR-300HP2 (mt7620)
I-O DATA WN-AC1167GR (mt7620)
I-O DATA WN-AC733GR3 (mt7620)

- mt7621
Buffalo WSR-1166DHP
ELECOM WRC-1167GHBK2-S (mt7621)
ELECOM WRC-2533GST (mt7621)
I-O DATA WN-AX1167GR (mt7621)
I-O DATA WN-GX300GR (mt7621)
Planex VR500 (mt7621)

- mt76x8
Tama W06

adschm committed:
ramips: Sort all subtarget Makefiles

This patch applies sorting to the definitions as whole blocks.

Sorting has been performed fully automatic, line count differences
originate from double empty lines removed.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adrianschmutzler commented:
Rebased due to DEVICE_TITLE split and to include #2170 (still in ynezz' staging tree).

Hope someone will eventually bring this to master...

dangowrt commented:
Merged. Thanks!
